### ISSUE TITLE
How to  check that feature vectors of the same person are close whereas those of different people are relatively far?

### ISSUE THREAD

**Issue by fabiozappo** [layumi/Person_reID_baseline_pytorch#276]
Hi @layumi and thank you for sharing your code in a very effective way!

This is my first time approaching person re-identification so forgive me if I'm referring to concepts with wrong terms.

My project consists in considering several lightweight architectures (efficientnet, mobilenet, squeezenet etc..), train them on Market and finally use them as feature extractor in order to reidentify each person. 

The goal is to evaluate for each architecture how good are the extracted features. By "how good" I mean that feature vectors of the same person are close whereas those of different people are relatively far. 

I think training with circle loss instead of a crossentropyloss helps the model to have a better feature separability, can you confirm it?

Are loss and top1error on validation set a good metric to my purpose? If not, does your repo contains a different way to see how much features vectors are well separated for each person?

**Comment by layumi** [comment]
Hi @fabiozappo 
I have compared different networks in my another arXiv paper https://arxiv.org/abs/2006.04569 , which are all trained from scratch for a fair comparison. 

I have done with the circle loss training and I will add more results to update the arXiv. 
Yes. Person re-id can be viewed as a metric learning approach. The evaluation metric is based on retrieving the right person of different view to the front of the ranking list.
